pyvrp/cpp/Route.cpp

```diff
@@ -221,49 +215,68 @@ Route::Route(ProblemData const &data, Trips trips, size_t vehType)
         auto const [x, y] = trip.centroid();
         centroid_.first += (x * trip.size()) / size();
         centroid_.second += (y * trip.size()) / size();
+    }
+
+    distanceCost_ = vehData.unitDistanceCost * static_cast<Cost>(distance_);
+    excessDistance_ = std::max<Distance>(distance_ - vehData.maxDistance, 0);
 
+    for (size_t idx = 0; idx != trips_.size(); ++idx)  // load statistics
+    {
+        auto const &trip = trips_[idx];
         auto const &tripDeliv = trip.delivery();
         auto const &tripPick = trip.pickup();
         auto const &tripLoad = trip.load();
+
         for (size_t dim = 0; dim != data.numLoadDimensions(); ++dim)
         {
             LoadSegment ls = {tripDeliv[dim], tripPick[dim], tripLoad[dim], 0};
 
-            if (tripIdx == 0 && vehData.initialLoad[dim] > 0)
-                // There is initial load that the first trip does not know
-                // about, but we need to account for that first.
+            if (idx == 0 && vehData.initialLoad[dim] > 0)
+                // This is initial load that the first trip does not know about
+                // that we need to account for first.
                 ls = LoadSegment::merge({vehData, dim}, ls);
 
             delivery_[dim] += ls.delivery();
             pickup_[dim] += ls.pickup();
             excessLoad_[dim] += ls.excessLoad(vehData.capacity[dim]);
         }
+    }
 
-        size_t prevClient = trip.startDepot();
-        for (auto const client : trip)
+    // Duration statistics. We iterate in reverse, that is, from the last to
```

Comment:
Question: why merge segments back-to-front instead of front-to-back? Is it because `finaliseFront` is faster than `finaliseBack`?

Reply:
Yes, and it's a bit easier at the start  - there's some statistics like `slack` and startTime that we track about the first trip/start of the route, that now no longer need a special check, we can just get them from the complete duration segment.